ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -155,8 +156,10 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, NetworkMetr
             connId = selector.connect(new InetSocketAddress(host, port.getPort()), networkConfig.socketSendBufferBytes,
                 networkConfig.socketReceiveBufferBytes, port.getPortType());
             connectionTracker.startTrackingInitiatedConnection(host, port, connId);
+            logger.trace("Initiated a connection to " + host + ":" + port);
```

Comment:
don't use concatenation for log statements. You will suffer the concatenation cost even though this is not printed most of the times because it is trace. This should be written as

`logger.trace("Initiated a connection to {}:{}", host, port);`

Same comment for all log statements.